crates/hir_ty/src/method_resolution.rs

```diff
@@ -304,7 +305,7 @@ pub(crate) fn lookup_method(
 pub enum LookupMode {
     /// Looking up a method call like `v.len()`: We only consider candidates
     /// that have a `self` parameter, and do autoderef.
-    MethodCall,
+    MethodCall { from_module: Option<ModuleId> },
```

Comment:
Not sure if this is a good idea?

Reply:
Yeah I don't know either, it make more sense to just pass a parameter `visible_from_module` around

Reply:
The thing is, doesn't passing a module id only makes sense for the MethodCall lookup mode? That's why I included it in the MethodCall variant. But I could also adjust this to just pass a visible_from_module around.

Reply:
Yeah, I just think it'll probably be simpler that way, and the places using path lookup can just not pass the module.